src/api/routes/rosetta/construction.ts

```diff
@@ -297,5 +324,174 @@ export function createRosettaConstructionRouter(db: DataStore): RouterWithAsync
     }
   });
 
+  //construction/payloads endpoint
+  router.postAsync('/payloads', async (req, res) => {
+    const valid: ValidSchema = await rosettaValidateRequest(req.originalUrl, req.body);
+    if (!valid.valid) {
+      res.status(400).json(makeRosettaError(valid));
+      return;
+    }
+
+    const options = getOptionsFromOperations(req.body.operations);
+    if (options == null) {
+      res.status(400).json(RosettaErrors.invalidOperation);
+      return;
+    }
+
+    const amount = options.amount;
+    if (!amount) {
+      res.status(400).json(RosettaErrors.invalidAmount);
+      return;
+    }
+
+    const fees = options.fee;
+    if (!fees) {
+      res.status(400).json(RosettaErrors.invalidFees);
+      return;
+    }
+
+    const publicKeys: RosettaPublicKey[] = req.body.public_keys;
+    if (!publicKeys) {
+      res.status(400).json(RosettaErrors.emptyPublicKey);
+      return;
+    }
+    for (const key of publicKeys) {
+      if (key.curve_type !== 'secp256k1') {
+        res.status(400).json(RosettaErrors.invalidCurveType);
+        return;
+      }
+    }
+
+    const recipientAddress = options.token_transfer_recipient_address;
+    if (!recipientAddress) {
+      res.status(400).json(RosettaErrors.invalidRecipient);
+      return;
+    }
+    const senderAddress = options.sender_address;
+
+    if (!senderAddress) {
+      res.status(400).json(RosettaErrors.invalidSender);
+      return;
+    }
+
+    const accountInfo = await new StacksCoreRpcClient().getAccount(senderAddress);
+
+    let tokenTransferOptions: UnsignedTokenTransferOptions | UnsignedMultiSigTokenTransferOptions;
+
+    if (publicKeys.length > 1) {
+      //multi signature
+      const publicKeysStrings = publicKeys.map(key => {
+        return key.hex_bytes;
+      });
+      tokenTransferOptions = {
+        recipient: recipientAddress,
+        amount: new BN(amount),
+        fee: new BN(fees),
+        publicKeys: publicKeysStrings,
+        numSignatures: 2,
```

Comment:
This appears to only allow for 2-of-2 multisig transactions, it should support N-of-M

Reply:
It should be equal to the number of public keys the user provides right?